### PR TITLE
HPCC-25885 Add WsDali methods for setprotect, unprotect, and listprotect

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

File: dali/daliadmin/daadmin.cpp [hunk 1]
```diff
@@ -1103,28 +1103,33 @@ int dfsverify(const char *name,CDateTime *cutoff, IUserDescriptor *user)
 
 //=============================================================================
 
-void setprotect(const char *filename, const char *callerid, IUserDescriptor *user)
+bool setprotect(const char *filename, const char *callerid, IUserDescriptor *user, StringBuffer &out)
```
Comment (wangkx) [inline dali/daliadmin/daadmin.cpp:1106]: You may keep the 'void' and remove the 'return true;' below -- similar to the 'setdfspartattr'.

File: dali/daliadmin/daadmin.cpp [hunk 2]
```diff
@@ -1103,28 +1103,33 @@ int dfsverify(const char *name,CDateTime *cutoff, IUserDescriptor *user)
...
     Owned<IDistributedFile> file = queryDistributedFileDirectory().lookup(filename,user,false,false,false,nullptr,defaultPrivilegedUser);
     file->setProtect(callerid,true);
+    out.appendf("%s is protected", file->queryLogicalName());
+    return true;
 }
 
 //=============================================================================
 
-void unprotect(const char *filename, const char *callerid, IUserDescriptor *user)
+bool unprotect(const char *filename, const char *callerid, IUserDescriptor *user, StringBuffer &out)
```
Comment (wangkx) [inline dali/daliadmin/daadmin.cpp:1116]: Same as setprotect.

File: dali/daliadmin/daadmin.cpp [hunk 3]
```diff
@@ -1103,28 +1103,33 @@ int dfsverify(const char *name,CDateTime *cutoff, IUserDescriptor *user)
...
 {
     Owned<IDistributedFile> file = queryDistributedFileDirectory().lookup(filename,user,false,false,false,nullptr,defaultPrivilegedUser);
     file->setProtect((strcmp(callerid,"*")==0)?NULL:callerid,false);
+    out.appendf("%s is unprotected", file->queryLogicalName());
+    return true;
 }
 //=============================================================================
 
-void listprotect(const char *filename, const char *callerid)
+bool listprotect(const char *filename, const char *callerid, StringBuffer &out)
```
Comment (wangkx) [inline dali/daliadmin/daadmin.cpp:1125]: Same as setprotect.

**Review by wangkx — COMMENTED** [review]
Looks good. 1 very minor comment.

File: dali/daliadmin/daadmin.cpp (hunk 1 above)
Comment (Michael-Gardner) [reply dali/daliadmin/daadmin.cpp]: Changed it back to void and fixed it in other areas.  Works just like setdfspartattr now.

File: dali/daliadmin/daadmin.cpp (hunk 2 above)
Comment (Michael-Gardner) [reply dali/daliadmin/daadmin.cpp]: Made same change as setprotect.

File: dali/daliadmin/daadmin.cpp (hunk 3 above)
Comment (Michael-Gardner) [reply dali/daliadmin/daadmin.cpp]: Made same change as setprotect.

**Review by wangkx — APPROVED** [review]
@Michael-Gardner looks fine

**Pull request merged**